torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -168,186 +156,32 @@ def __init__(
         neighbor_sampler: Optional[NeighborSampler] = None,
         **kwargs,
     ):
-        # Remove for PyTorch Lightning:
-        if 'dataset' in kwargs:
-            del kwargs['dataset']
-        if 'collate_fn' in kwargs:
-            del kwargs['collate_fn']
-
-        self.data = data
+        # Get input type:
+        # TODO(manan): this computation is required twice, once here and once
+        # in LinkLoader:
+        edge_type, _ = get_edge_label_index(data, edge_label_index)
```

Comment:
Yeah, what are ways to resolve this? Move this to a property of the loader?

Reply:
Yeah, or split up the logic to obtain an edge type and edge label index. Will handle in a follow-up :)